packages/hash/frontend/src/blocks/page/InsertBlock.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, FunctionComponent } from "react";
+import { Box, popoverClasses } from "@mui/material";
+import { Popover } from "@hashintel/hash-design-system";
+import { HashBlockMeta } from "@hashintel/hash-shared/blocks";
+import type { BlockVariant } from "@blockprotocol/core";
+import { PlusBoxOutlineIcon } from "../../shared/icons/plus-box-outline-icon";
+import { BlockSuggester } from "./createSuggester/BlockSuggester";
+
+type InsertBlockProps = {
+  onBlockSuggesterChange: (
+    variant: BlockVariant,
+    blockMeta: HashBlockMeta,
+  ) => void;
+};
+
+export const InsertBlock: FunctionComponent<InsertBlockProps> = ({
+  onBlockSuggesterChange,
+}) => {
+  const [contextMenu, setContextMenu] = useState<{
```

Comment:
Nit: I think we should call this contextMenuPosition for more clarity

Reply:
Done in 7f45bcd0b2619a71d3e5d1c378902803c1cddf97